src/bolos/cx_ec_domain.c

```diff
@@ -1399,6 +1482,81 @@ int cx_nid_from_curve(cx_curve_t curve)
   return nid;
 }
 
+EC_GROUP *cx_group_from_nid_and_curve(int nid, cx_curve_t cid)
+{
+  EC_GROUP *group = NULL;
+  /* create a generic curve if generic nid */
+  if (nid != NID_undef) {
+    group = EC_GROUP_new_by_curve_name(nid);
+  } else {
+    group = cx_create_generic_curve(cid);
+  }
+  return group;
+}
+
+EC_GROUP *cx_create_generic_curve(cx_curve_t cid)
+{
+  /* Get curve domain to init the EC_GROUP */
+  const cx_curve_domain_t *domain;
+  if (NULL == (domain = cx_ecdomain(cid))) {
+    errx(1, "error when getting the curve domain");
+    goto error;
+  }
+
+  BN_CTX *ctx;
+  EC_GROUP *curve;
```

Comment:
I think `curve` should be initialized to `NULL`, I don't know why the compiler doesn't complain.

Reply:
ACK